fud2/fud-core/src/exec/plan/json_planner.rs

```diff
@@ -2,30 +2,60 @@
 //! and must be special cased by the logic later. Currently planners only emit states with file
 //! assignment done later.
 
-use super::{FindPlan, PlannerType};
+use std::io::{self, Read as _};
+
+use cranelift_entity::PrimaryMap;
+
+use crate::{
+    exec::{OpRef, Operation, State, StateRef},
+    flang::ast_to_ir,
+};
+
+use super::{FindPlan, PlanReq, planner::PlanResp};
 
 #[derive(Debug)]
 pub struct JsonPlanner {}
 
 impl FindPlan for JsonPlanner {
     fn find_plan(
         &self,
-        _start: &[crate::exec::StateRef],
-        _end: &[crate::exec::StateRef],
-        _through: &[crate::exec::OpRef],
-        _ops: &cranelift_entity::PrimaryMap<
-            crate::exec::OpRef,
-            crate::exec::Operation,
-        >,
-        _states: &cranelift_entity::PrimaryMap<
-            crate::exec::StateRef,
-            crate::exec::State,
-        >,
-    ) -> Option<Vec<super::Step>> {
-        None
-    }
+        req: &PlanReq,
+        ops: &PrimaryMap<OpRef, Operation>,
+        _states: &PrimaryMap<StateRef, State>,
+    ) -> Option<PlanResp> {
+        let _ = _states;
+        let mut stdin = io::stdin().lock();
+        let mut input = String::new();
+        let res = stdin.read_to_string(&mut input);
+        if let Err(e) = res {
+            eprintln!("error: {e}");
+            return None;
+        }
 
-    fn ty(&self) -> PlannerType {
-        PlannerType::FromJson
+        let ast = &serde_json::from_str(&input);
+        match ast {
+            Err(e) => unimplemented!("{e}"),
```

Comment:
Maybe just `panic!` or `return None`? Can't think of anything that we _would_ want to implement in the case where the JSON is malformed.